src/Sylius/Bundle/CoreBundle/Processor/ProductCatalogPromotionsProcessor.php

```diff
@@ -13,44 +13,30 @@
 
 namespace Sylius\Bundle\CoreBundle\Processor;
 
-use Sylius\Bundle\CoreBundle\Applicator\CatalogPromotionApplicatorInterface;
-use Sylius\Bundle\PromotionBundle\Provider\EligibleCatalogPromotionsProviderInterface;
-use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Bundle\CoreBundle\Announcer\BatchedVariantsUpdateAnnouncerInterface;
 use Sylius\Component\Core\Model\ProductInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
-use Sylius\Component\Core\Provider\CatalogPromotionVariantsProviderInterface;
-use Sylius\Component\Resource\Repository\RepositoryInterface;
+
 
 final class ProductCatalogPromotionsProcessor implements ProductCatalogPromotionsProcessorInterface
 {
-    private EligibleCatalogPromotionsProviderInterface $catalogPromotionsProvider;
-
-    private CatalogPromotionClearerInterface $catalogPromotionClearer;
-
-    private CatalogPromotionVariantsProviderInterface $catalogPromotionVariantsProvider;
-
-    private CatalogPromotionApplicatorInterface $catalogPromotionApplicator;
-
     public function __construct(
-        EligibleCatalogPromotionsProviderInterface $catalogPromotionsProvider,
-        CatalogPromotionClearerInterface $catalogPromotionClearer,
-        CatalogPromotionVariantsProviderInterface $catalogPromotionVariantsProvider,
-        CatalogPromotionApplicatorInterface $catalogPromotionApplicator
+        private CatalogPromotionClearerInterface $catalogPromotionClearer,
+        private BatchedVariantsUpdateAnnouncerInterface $announcer
     ) {
-        $this->catalogPromotionsProvider = $catalogPromotionsProvider;
-        $this->catalogPromotionClearer = $catalogPromotionClearer;
-        $this->catalogPromotionVariantsProvider = $catalogPromotionVariantsProvider;
-        $this->catalogPromotionApplicator = $catalogPromotionApplicator;
     }
 
     public function process(ProductInterface $product): void
     {
         $variants = $product->getVariants()->toArray();
         $this->clearVariants($variants);
```

Comment:
I'm not sure if it is within the scope of this iteration, but I would say that clearing of previous promotions should be done during processing batched variants. This way we won't have to remember about it before update execution